pkg/sources/adapter/cloudeventssource/cloudevents_test.go

```diff
@@ -0,0 +1,223 @@
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Contains code excerpt based on VMware's VEBA's webhook
+
+vCenter Event Broker Appliance
+Copyright (c) 2019 VMware, Inc.  All rights reserved
+
+The BSD-2 license (the "License") set forth below applies to all parts of the vCenter Event Broker Appliance project.  You may not use this file except in compliance with the License.
+
+BSD-2 License
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+
+package cloudeventssource
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	zapt "go.uber.org/zap/zaptest"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	cloudeventst "github.com/cloudevents/sdk-go/v2/client/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/triggermesh/triggermesh/pkg/adapter/fs"
+)
+
+const (
+	// path to secret fixtures.
+	secret1Path = "../../../../test/fixtures/secrets/secret1"
+	secret2Path = "../../../../test/fixtures/secrets/secret2"
+
+	// values for secrets in the fixtures above.
+	tSecret1 = "secret1"
+	tSecret2 = "secret2"
+
+	tUser  = "user"
+	tToken = "token"
+)
+
+var (
+	basicAuths KeyMountedValues = KeyMountedValues{
+		{
+			Key:              tUser,
+			MountedValueFile: secret1Path,
+		},
+	}
+	tokens KeyMountedValues = KeyMountedValues{
+		{
+			Key:              tToken,
+			MountedValueFile: secret2Path,
+		},
+	}
+)
+
+func TestCloudEventsSource(t *testing.T) {
+	logger := zapt.NewLogger(t).Sugar()
```

Comment:
nit: Knative has a `logging/testing` package, any reason not to use it here?

https://github.com/triggermesh/triggermesh/blob/f2928696c9d11bdc1100b0d153cfb7c0547ac4bd/pkg/sources/adapter/awscodecommitsource/adapter_test.go#L69

Some stats:

```console
$ grep -rHn 'NewLogger' pkg/sources/adapter | wc -l
3
```

```console
$ grep -rHn 'TestLogger' pkg/sources/adapter | wc -l
28
```

Trying to see if we can do this in a consistent way, and more importantly reuse what Knative has to offer.

Reply:
https://github.com/triggermesh/triggermesh/pull/730/commits/64ac555e2f9c7c4a35d31f2515e79c6bff6da83c